OutOfSchool/OutOfSchool.DataAccess/Repository/PhotoRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace OutOfSchool.Services.Repository
+{
+    public class PhotoRepository : IPhotoRepository
+    {
+        private readonly string basePhotoPath;
+
+        public PhotoRepository(IConfiguration config)
+        {
+            basePhotoPath = config.GetSection("PhotoSettings:BasePath").Value;
+        }
+
+        /// <summary>
+        /// Create photo.
+        /// </summary>
+        /// <param name="photo">Photo as byte array.</param>
+        /// <param name="fileName">Photo name.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task CreateUpdatePhoto(byte[] photo, string fileName)
+        {
+            var filePath = GenerateFilePath(fileName);
+
+            Directory.CreateDirectory(Path.GetDirectoryName(filePath));
```

Comment:
According to the information from the client, we do not  know yet where the files will be stored. (Probably, it will be NAS / NFS). When the information will be in more detail, we can easily add another implementation to the interface.

Reply:
Backup, moving data to other server, access rights, ... - for case of separate folder has higher maintainability cost.